rclcpp/include/rclcpp/experimental/buffers/ring_buffer_implementation.hpp

```diff
@@ -112,7 +131,7 @@ class RingBufferImplementation : public BufferImplementationBase<BufferT>
   size_t read_index_;
   size_t size_;
 
-  std::mutex mutex_;
+  mutable std::shared_timed_mutex mutex_;
```

Comment:
are we going to use timeout for lock in the future? could you explain why you pick the `shared_timed_mutex` instead of `shared_mutex`?

Reply:
Good question.  `rclcpp` defaults to using C++14, and in C++14 `shared_mutex` was renamed to `shared_timed_mutex` since it had timing in it.  `shared_mutex` is available from C++17 onwards. (See [here](https://stackoverflow.com/a/40207246/3335057) or [here](https://stackoverflow.com/a/42302130/3335057).)